docs/validator.go

```diff
@@ -0,0 +1,167 @@
+// +build ignore
+
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage
+ * (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/a8m/mark"
+	"github.com/gernest/wow"
+	"github.com/gernest/wow/spin"
+	"github.com/minio/cli"
+)
+
+func init() {
+	// Validate go binary.
+	if _, err := exec.LookPath("go"); err != nil {
+		panic(err)
+	}
+}
+
+var globalFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "m",
+		Value: "API.md",
+		Usage: "Path to markdown api documentation.",
+	},
+	cli.StringFlag{
+		Name:  "t",
+		Value: "checker.go.tpl",
+		Usage: "Template used for generating the programs.",
+	},
+	cli.IntFlag{
+		Name:  "skip",
+		Value: 2,
+		Usage: "Skip entries before validating the code.",
+	},
```

Comment:
I think we would need to have a template like `mc --help` to show how this can be used with examples.

Reply:
Will add that.